### ISSUE TITLE
Tree keyboard navigation doesn't work on Safari

### ISSUE THREAD

**Issue by jamalsa** [mantinedev/mantine#9045]
### Dependencies check up

- [x] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

9.4.1

### What package has an issue?

@mantine/core

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

Safari

### Describe the bug

Keyboard navigation doesn't work on Safari both on basic and virtualized one.

Steps to reproduce:
1. Open `https://mantine.dev/core/tree/` on Safari
2. Click on a sample tree root
3. Navigate using arrow down/arrow up tree

Expectation: focus box will be visible and move according to keyboard event
Reality: Nothing happen when pressing arrow up/down key. But when pressing arrow right key the focus box visible

The navigation works without problem on Chrome

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by saadpocalypse** [comment]
I've just submitted a [PR](https://github.com/mantinedev/mantine/pull/9049) to resolve this.
